### PR TITLE
fix(parser): accept unquoted space-concat in field keys per S10.8 (#66)

### LINKED ISSUE #66: Spec violation: S10.8 — unquoted space-concat key  must be accepted
## Spec violation: S10.8

**HOCON spec reference:** §Value concatenation L317; §Paths as keys L553-560

> Because path expressions work like value concatenations, you can have whitespace in keys:
> `a b c : 42` is equivalent to `"a b c" : 42`

**Current behavior:** `foo bar = 42` is rejected as a parse error. Quoted form `"foo bar" = 42` works correctly (✅).

**Expected behavior:** unquoted space-separated tokens forming a single path element (e.g. `a b c : 42`) must be accepted and stored under the single key `"a b c"`.

**Regression pin:** `s10_8_unquoted_space_key_pin` in `tests/integration_test.rs` (Phase 2 block).
**Spec test:** `s10_8_unquoted_space_key_spec` (currently `#[ignore]`).

Introduced in Phase 2 spec compliance audit.

### PR DESCRIPTION
**Pull request by y1o1**
## Summary

- Implements S10.8 (HOCON spec L317 + L553-560) — `foo bar = 1` now parses as key `"foo bar"` instead of erroring with `unexpected token after key: Unquoted`.
- `parse_key` gains a space-concat continuation branch via a `space_concat` flag: when the next key token has `preceding_space=true`, the first dot-split piece merges into the LAST existing segment with a literal space; any remaining pieces become new path segments.
- Leading '.' after whitespace stays a path separator per S11.1 (e.g. `a .b` → `["a", "b"]`, NOT `["a b"]`). This edge case was flagged on the ts.hocon companion PR by Claude + Codex + Copilot independent review (3-way convergence); the rs port preserves the same semantics.
- 12 regression tests pin: basic 2-token, 3-token spec L556 example, dotted-prefix + dotted-tail concat, quoted+unquoted concat, inline-object shorthand, all 4 leading-dot edge cases, and tab whitespace.

## Behaviour examples (all pinned by tests)

| Input | Key (post-fix) |
|---|---|
| `foo bar = 1` | `"foo bar"` |
| `a b c : 42` | `"a b c"` (spec L556) |
| `a.b c = 1` | `["a", "b c"]` |
| `a b.c = 1` | `["a b", "c"]` |
| `"foo bar" baz = 1` | `"foo bar baz"` |
| `a b { x = 1 }` | key `"a b"`, value the inline object |
| `a .b = 1` | `["a", "b"]` (leading dot stays separator per S11.1) |
| `a .b.c = 1` | `["a", "b", "c"]` |
| `"a" .b = 1` | `["a", "b"]` |
| `a.b .c = 1` | `["a", "b", "c"]` |
| `"a.b" .c = 1` | `["a.b", "c"]` |
| `a\tb = 1` | `"a b"` (tab counts as whitespace) |

## Cross-impl

- [x] ts.hocon [#128](https://github.com/o3co/ts.hocon/pull/128) — MERGED
- [ ] **this PR (rs.hocon #66)**
- [ ] go.hocon #65 — pending after this lands

xx.hocon compliance-matrix update will be a separate PR after all 3 impls ship.

Closes #66.

## Test plan

- [x] `cargo test --workspace` — all 38 test binaries pass (122 integration tests including 13 S10.8)
- [x] `cargo test --workspace --features serde` — clean
- [x] `cargo clippy --all-targets --all-features -- -D warnings` — clean
- [x] No regression on existing key tests (`a.b = 1`, `"a.b" = 1`, `a."b.c".d = 1`, `a. b` trailing-dot continuation)
- [x] Value-position concat (`x = foo bar`) unaffected — change is localised to `parse_key`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/integration_test.rs [hunk 1]
```diff
@@ -743,8 +743,8 @@ fn s10_7_concat_does_not_span_newline() {
 
 // --- S10.8: string concat allowed in field keys (HOCON L317) -----------------
 // Spec L553-560: path expressions work like value concatenations, so
-// `a b c : 42` is a single-element path with key "a b c". ✅ for quoted keys;
-// unquoted space-separated keys are ❌.
+// `a b c : 42` is a single-element path with key "a b c". Both quoted and
+// unquoted space-separated forms must be accepted (#66 fixed in v1.5.0).
```
[the remark this answers is not in the document]
Comment (y1o1) [reply tests/integration_test.rs]: Fixed in e0e01ea — dropped the `v1.5.0` forward reference (Cargo.toml is still 1.4.1; the entry sits under `[Unreleased]` in CHANGELOG). Claude internal review also flagged the same label-drift risk → convergent fix.

File: docs/spec-compliance.md [hunk 2]
```diff
@@ -218,8 +218,8 @@ same item descriptions verbatim.
   tests: tests/integration_test.rs:689 (s10_7_concat_does_not_span_newline)
   status: ✅
 - **S10.8** String concat allowed in field keys — §Value concatenation (L317)
-  tests: tests/integration_test.rs:711 (s10_8_quoted_key_with_space_allowed); tests/integration_test.rs:722 (s10_8_unquoted_space_key_pin); tests/integration_test.rs:733 (s10_8_unquoted_space_key_spec)
-  status: ⚠️ quoted form ✅; unquoted space-concat key ❌ (see #66)
+  tests: tests/integration_test.rs (s10_8_* it blocks: quoted, basic, three-token, dotted-prefix concat, dotted-tail concat, quoted+unquoted, inline-object shorthand, leading-dot + S11.1 interaction, tab whitespace)
```
[the remark this answers is not in the document]
Comment (y1o1) [reply docs/spec-compliance.md]: Fixed in e0e01ea — rephrased "s10_8_* it blocks" → "s10_8_* tests". The `it()` jargon was carried over from the ts.hocon companion doc (vitest); rs uses `#[test]`.
